file.go

```diff
@@ -0,0 +1,194 @@
+package mset
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	catalogBaseDir        = ".m2"
+	catalogPathEnvVar     = "MSET_CATALOG_PATH" // replace default e.g ~/myDirectory
+	mavenPathEnvVar       = "MSET_MAVEN_PATH"
+	mavenSettingsFileName = "settings.xml"
+	catalogDirectoryName  = ".mset"
+	fileSuffix            = "-settings.xml"
+	fileEntryRegexp       = `^[a-zA-Z0-9-]*$`
+	configFileCurrent     = ".mcurrent"
+)
+
+// check if the file exist
+func fileExist(path string) error {
+	fi, err := pathExist(path)
+	if err != nil {
+		return err
+	}
+
+	if fi.IsDir() {
+		return fmt.Errorf("path '%v' is a directory", path)
+	}
+	return nil
+}
+
+func directoryExist(path string) error {
+	fi, err := pathExist(path)
+	if err != nil {
+		return err
+	}
+
+	if !fi.IsDir() {
+		return fmt.Errorf("path '%v' is a file", path)
+	}
+	return nil
+}
+
+func pathExist(path string) (os.FileInfo, error) {
+	fi, err := os.Stat(path)
+
+	if os.IsNotExist(err) {
+		return nil, err
+	}
+	return fi, nil
+}
+
+//
+func isValidName(name string) bool {
```

Comment:
I don't see where is this func being used besides the test

Reply:
ups I'm blind, I'll say that if you are using MustCompile you can declare it outside  and not compiling it each time the func is called

Reply:
In this case I can create a global variable, the cool thing is regexp is thread safe

Reply:
I found this, In this case MustCompile is not necessary
```
func isValidName(name string) bool {
	match, _ := regexp.MatchString(fileEntryRegexp, name)
	return match
}
```
What do you think? better?